### ISSUE TITLE
Unable to run the blueriver-muracms-mysql example

### ISSUE THREAD

**Issue by SowjiChenji** [blueriver/docker-muracms#3]
![image](https://user-images.githubusercontent.com/49544377/80551837-b0e3f900-8979-11ea-9efb-5494fdb65228.png)

It takes forever to download.

**Comment by umbraesoulsbane** [comment]
Pretty sure those images both come from dockerhub so you might want to test your connection to there. The MySQL image definely isn't maintained by Bluer River. Images can take a while on first fun so long delays would be expected normally; especially over wifi.

If it's hanging on wait longer than the last chunks you might also want to try deleting the image and cache you have now and redownload it from scratch. I have had parts get corrupt and hang up that way.

**Comment by SowjiChenji** [comment]
Oh yeah, my bad! I just realized it was the MySQL image. However, the issue was with docker freezing while downloading multiple images at once due to my low bandwidth. I had to reduce the maximum concurrent downloads for docker. 

`https://stackoverflow.com/a/53915667`

> {
    "max-concurrent-uploads": 1,
    "max-concurrent-downloads": 1
}

**Comment by SowjiChenji** [comment]
I was able to finally get the containers up and running but I think there is an issue connecting to the MySQL container. @umbraesoulsbane ,  Any thoughts? 

`
500 Error
Communications link failure The last packet successfully received from the server was 42 milliseconds ago. The last packet sent successfully to the server was 37 milliseconds ago.
Code:0
Type:database
lucee.runtime.exp.DatabaseException: Communications link failure

The last packet successfully received from the server was 42 milliseconds ago.  The last packet sent successfully to the server was 37 milliseconds ago.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:989)
	at com.mysql.jdbc.ExportControlled.transformSocketToSSLSocket(ExportControlled.java:164)
	at com.mysql.jdbc.MysqlIO.negotiateSSLConnection(MysqlIO.java:4898)
	at com.mysql.jdbc.MysqlIO.proceedHandshakeWithPluggableAuthentication(MysqlIO.java:1659)
	at com.mysql.jdbc.MysqlIO.doHandshake(MysqlIO.java:1226)
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2253)
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2284)
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2083)
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:806)
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:410)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:328)
	at lucee.runtime.db.DataSourceSupport._getConnection(DataSourceSupport.java:106)
	at lucee.runtime.db.DataSourceSupport.getConnection(DataSourceSupport.java:86)
	at lucee.runtime.db.DatasourceConnectionPool.loadDatasourceConnection(DatasourceConnectionPool.java:141)
	at lucee.runtime.db.DatasourceConnectionPool.getDatasourceConnection(DatasourceConnectionPool.java:105)
	at lucee.runtime.db.DatasourceManagerImpl.getConnection(DatasourceManagerImpl.java:75)
	at lucee.runtime.db.DatasourceManagerImpl.getConnection(DatasourceManagerImpl.java:69)
	at lucee.runtime.tag.DBInfo.doStartTag(DBInfo.java:258)
	at core.appcfc.setup_check_cfm$cf.call(/muraWRM/core/appcfc/setup_check.cfm:6)
	at lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:933)
	at lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:823)
	at lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:805)
	at core.appcfc.onapplicationstart_include_cfm$cf.call(/muraWRM/core/appcfc/onApplicationStart_include.cfm:74)
	at lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:933)
	at lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:823)
	at lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:805)
	at core.appcfc.onapplicationstart_method_cfm$cf.udfCall(/core/appcfc/onApplicationStart_method.cfm:49)
	at lucee.runtime.type.UDFImpl.implementation(UDFImpl.java:107)
	at lucee.runtime.type.UDFImpl._call(UDFImpl.java:357)
	at lucee.runtime.type.UDFImpl.call(UDFImpl.java:226)
	at lucee.runtime.ComponentImpl._call(ComponentImpl.java:693)
	at lucee.runtime.ComponentImpl._call(ComponentImpl.java:573)
	at lucee.runtime.ComponentImpl.call(ComponentImpl.java:1997)
	at lucee.runtime.listener.ModernAppListener.call(ModernAppListener.java:438)
	at lucee.runtime.listener.ModernAppListener.onApplicationStart(ModernAppListener.java:306)
	at lucee.runtime.PageContextImpl.initApplicationContext(PageContextImpl.java:3113)
	at lucee.runtime.listener.ModernAppListener._onRequest(ModernAppListener.java:118)
	at lucee.runtime.listener.MixedAppListener.onRequest(MixedAppListener.java:43)
	at lucee.runtime.PageContextImpl.execute(PageContextImpl.java:2464)
	at lucee.runtime.PageContextImpl._execute(PageContextImpl.java:2454)
	at lucee.runtime.PageContextImpl.executeCFML(PageContextImpl.java:2427)
	at lucee.runtime.engine.Request.exe(Request.java:44)
	at lucee.runtime.engine.CFMLEngineImpl._service(CFMLEngineImpl.java:1090)
	at lucee.runtime.engine.CFMLEngineImpl.serviceCFML(CFMLEngineImpl.java:1038)
	at lucee.loader.engine.CFMLEngineWrapper.serviceCFML(CFMLEngineWrapper.java:102)
	at lucee.loader.servlet.CFMLServlet.service(CFMLServlet.java:51)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.filters.ExpiresFilter.doFilter(ExpiresFilter.java:1225)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176)
	at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145)
	at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92)
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:394)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:94)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:492)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:684)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:502)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1152)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)
	at org.apache.tomcat.util.net.AprEndpoint$SocketWithOptionsProcessor.run(AprEndpoint.java:2464)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)

Column: 0
ID: ??
Line: 6
Raw Trace: core.appcfc.setup_check_cfm$cf.call(/muraWRM/core/appcfc/setup_check.cfm:6)
Template: /var/www/core/appcfc/setup_check.cfm
Type: cfml

Column: 0
ID: ??
Line: 74
Raw Trace: core.appcfc.onapplicationstart_include_cfm$cf.call(/muraWRM/core/appcfc/onApplicationStart_include.cfm:74)
Template: /var/www/core/appcfc/onApplicationStart_include.cfm
Type: cfml

Column: 0
ID: ??
Line: 49
Raw Trace: core.appcfc.onapplicationstart_method_cfm$cf.udfCall(/core/appcfc/onApplicationStart_method.cfm:49)
Template: /var/www/core/appcfc/onApplicationStart_method.cfm
Type: cfml
`

**Comment by umbraesoulsbane** [comment]
I believe that is just a first time run timeout. At setup Mura will start creating the DB structure in the MySQL database. Depending on the system resources this can take a while and Lucee can timeout.

Spin up the containers again. If you get that error give a little time and then put ?appreload at the end of the url and hit enter. Appreload will cause Mura to try again which includes checking the configuration and db setup. If the creation process completed successfully then it should recognize that and finish the installation process.

It may be possible that the DB creation process corrupted something if you stopped the containers before the DB was created. If so, you can delete the volume vol_muradb which will completely delete the database to allow everything to be recreated from scratch.

**Comment by SowjiChenji** [comment]
It still doesn't help ! I waited for a while restarted containers. Tried a couple of times and then recreated the containers but din't work. Does the below log help?

`"ERROR","http-apr-8888-exec-7","05/01/2020","02:07:12","","{""Extended_Info"":"""",""Message"":""Page /config/cfapplication.cfm [/var/www/config/cfapplication.cfm] not found"",""Detail"":"""",""MissingFileName_abs"":""/var/www/config/cfapplication.cfm"",""additional"":{""Mapping"":""/""},""MissingFileName_rel"":""/config/cfapplication.cfm"",""TagContext"":[{""Raw_Trace"":""core.appcfc.applicationsettings_cfm$cf.call(/core/appcfc/applicationSettings.cfm:304)"",""codePrintPlain"":""302: this.ormSettings.cfclocation=[];\n303: try {\n304: \tinclude \""#variables.context#/config/cfapplication.cfm\"";\n305: \trequest.hasCFApplicationCFM=true;\n306: } catch (any e) {\n"",""column"":0,""line"":304,""template"":""/var/www/core/appcfc/applicationSettings.cfm"",""id"":""??"",""type"":""cfml"",""codePrintHTML"":""302: this.ormSettings.cfclocation=[];<br>\n303: try {<br>\n<b>304: \tinclude &quot;#variables.context#/config/cfapplication.cfm&quot;;</b><br>\n305: \trequest.hasCFApplicationCFM=true;<br>\n306: } catch (any e) {<br>\n""},{""Raw_Trace"":""application_cfc$cf.initComponent(/Application.cfc:46)"",""codePrintPlain"":""44: component output=\""false\"" {\n45: \n46: \tinclude \""core/appcfc/applicationSettings.cfm\"";\n47: \n48: \n"",""column"":0,""line"":46,""template"":""/var/www/Application.cfc"",""id"":""??"",""type"":""cfml"",""codePrintHTML"":""44: component output=&quot;false&quot; {<br>\n45: <br>\n<b>46: \tinclude &quot;core/appcfc/applicationSettings.cfm&quot;;</b><br>\n47: <br>\n48: <br>\n""},{""Raw_Trace"":""application_cfc$cf.newInstance(/Application.cfc:44)"",""codePrintPlain"":""42: version 2 without this exception.  You may, if you choose, apply this exception to your own modified versions of Mura CMS.\n43: */\n44: component output=\""false\"" {\n45: \n46: \tinclude \""core/appcfc/applicationSettings.cfm\"";\n"",""column"":0,""line"":44,""template"":""/var/www/Application.cfc"",""id"":""??"",""type"":""cfml"",""codePrintHTML"":""42: version 2 without this exception.  You may, if you choose, apply this exception to your own modified versions of Mura CMS.<br>\n43: */<br>\n<b>44: component output=&quot;false&quot; {</b><br>\n45: <br>\n46: \tinclude &quot;core/appcfc/applicationSettings.cfm&quot;;<br>\n""}],""ErrorCode"":""0"",""MissingFileName"":""/config/cfapplication.cfm"",""type"":""missinginclude"",""StackTrace"":""lucee.runtime.exp.MissingIncludeException: Page /config/cfapplication.cfm [/var/www/config/cfapplication.cfm] not found\n\tat lucee.runtime.PageSourceImpl.loadPage(PageSourceImpl.java:964)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:927)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:823)\n\tat lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:805)\n\tat core.appcfc.applicationsettings_cfm$cf.call(/core/appcfc/applicationSettings.cfm:304)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:933)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:823)\n\tat lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:805)\n\tat application_cfc$cf.initComponent(/Application.cfc:46)\n\tat application_cfc$cf.newInstance(/Application.cfc:44)\n\tat lucee.runtime.component.ComponentLoader.initComponent(ComponentLoader.java:556)\n\tat lucee.runtime.component.ComponentLoader._loadComponent(ComponentLoader.java:493)\n\tat lucee.runtime.component.ComponentLoader.loadComponent(ComponentLoader.java:377)\n\tat lucee.runtime.listener.ModernAppListener._onRequest(ModernAppListener.java:112)\n\tat lucee.runtime.listener.MixedAppListener.onRequest(MixedAppListener.java:43)\n\tat lucee.runtime.PageContextImpl.execute(PageContextImpl.java:2464)\n\tat lucee.runtime.PageContextImpl._execute(PageContextImpl.java:2454)\n\tat lucee.runtime.PageContextImpl.executeCFML(PageContextImpl.java:2427)\n\tat lucee.runtime.engine.Request.exe(Request.java:44)\n\tat lucee.runtime.engine.CFMLEngineImpl._service(CFMLEngineImpl.java:1090)\n\tat lucee.runtime.engine.CFMLEngineImpl.serviceCFML(CFMLEngineImpl.java:1038)\n\tat lucee.loader.engine.CFMLEngineWrapper.serviceCFML(CFMLEngineWrapper.java:102)\n\tat lucee.loader.servlet.CFMLServlet.service(CFMLServlet.java:51)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:729)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.apache.catalina.filters.ExpiresFilter.doFilter(ExpiresFilter.java:1225)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176)\n\tat org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145)\n\tat org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92)\n\tat org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:394)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)\n\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:94)\n\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:492)\n\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)\n\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)\n\tat org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)\n\tat org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:684)\n\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)\n\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:502)\n\tat org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1152)\n\tat org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)\n\tat org.apache.tomcat.util.net.AprEndpoint$SocketWithOptionsProcessor.run(AprEndpoint.java:2464)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)\n\tat java.lang.Thread.run(Thread.java:748)\n"",""ExtendedInfo"":""""}"
muracms_1     | "ERROR","http-apr-8888-exec-7","05/01/2020","02:07:13","","{""MESSAGE"":""Communications link failure\n\nThe last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server."",""STACKTRACE"":""lucee.runtime.exp.DatabaseException: Communications link failure\n\nThe last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.\n\tat sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n\tat sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)\n\tat sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)\n\tat java.lang.reflect.Constructor.newInstance(Constructor.java:423)\n\tat com.mysql.jdbc.Util.handleNewInstance(Util.java:425)\n\tat com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:989)\n\tat com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:341)\n\tat com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2251)\n\tat com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2284)\n\tat com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2083)\n\tat com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:806)\n\tat com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47)\n\tat sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n\tat sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)\n\tat sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)\n\tat java.lang.reflect.Constructor.newInstance(Constructor.java:423)\n\tat com.mysql.jdbc.Util.handleNewInstance(Util.java:425)\n\tat com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:410)\n\tat com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:328)\n\tat lucee.runtime.db.DataSourceSupport._getConnection(DataSourceSupport.java:106)\n\tat lucee.runtime.db.DataSourceSupport.getConnection(DataSourceSupport.java:86)\n\tat lucee.runtime.db.DatasourceConnectionPool.loadDatasourceConnection(DatasourceConnectionPool.java:141)\n\tat lucee.runtime.db.DatasourceConnectionPool.getDatasourceConnection(DatasourceConnectionPool.java:105)\n\tat lucee.runtime.db.DatasourceManagerImpl.getConnection(DatasourceManagerImpl.java:75)\n\tat lucee.runtime.db.DatasourceManagerImpl.getConnection(DatasourceManagerImpl.java:69)\n\tat lucee.runtime.tag.DBInfo.doStartTag(DBInfo.java:258)\n\tat core.appcfc.setup_check_cfm$cf.call(/muraWRM/core/appcfc/setup_check.cfm:6)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:933)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:823)\n\tat lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:805)\n\tat core.appcfc.onapplicationstart_include_cfm$cf.call(/muraWRM/core/appcfc/onApplicationStart_include.cfm:74)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:933)\n\tat lucee.runtime.PageContextImpl._doInclude(PageContextImpl.java:823)\n\tat lucee.runtime.PageContextImpl.doInclude(PageContextImpl.java:805)\n\tat core.appcfc.onapplicationstart_method_cfm$cf.udfCall(/core/appcfc/onApplicationStart_method.cfm:49)\n\tat lucee.runtime.type.UDFImpl.implementation(UDFImpl.java:107)\n\tat lucee.runtime.type.UDFImpl._call(UDFImpl.java:357)\n\tat lucee.runtime.type.UDFImpl.call(UDFImpl.java:226)\n\tat lucee.runtime.ComponentImpl._call(ComponentImpl.java:693)\n\tat lucee.runtime.ComponentImpl._call(ComponentImpl.java:573)\n\tat lucee.runtime.ComponentImpl.call(ComponentImpl.java:1997)\n\tat lucee.runtime.listener.ModernAppListener.call(ModernAppListener.java:438)\n\tat lucee.runtime.listener.ModernAppListener.onApplicationStart(ModernAppListener.java:306)\n\tat lucee.runtime.PageContextImpl.initApplicationContext(PageContextImpl.java:3113)\n\tat lucee.runtime.listener.ModernAppListener._onRequest(ModernAppListener.java:118)\n\tat lucee.runtime.listener.MixedAppListener.onRequest(MixedAppListener.java:43)\n\tat lucee.runtime.PageContextImpl.execute(PageContextImpl.java:2464)\n\tat lucee.runtime.PageContextImpl._execute(PageContextImpl.java:2454)\n\tat lucee.runtime.PageContextImpl.executeCFML(PageContextImpl.java:2427)\n\tat lucee.runtime.engine.Request.exe(Request.java:44)\n\tat lucee.runtime.engine.CFMLEngineImpl._service(CFMLEngineImpl.java:1090)\n\tat lucee.runtime.engine.CFMLEngineImpl.serviceCFML(CFMLEngineImpl.java:1038)\n\tat lucee.loader.engine.CFMLEngineWrapper.serviceCFML(CFMLEngineWrapper.java:102)\n\tat lucee.loader.servlet.CFMLServlet.service(CFMLServlet.java:51)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:729)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.apache.catalina.filters.ExpiresFilter.doFilter(ExpiresFilter.java:1225)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176)\n\tat org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145)\n\tat org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92)\n\tat org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:394)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)\n\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)\n\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:94)\n\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:492)\n\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)\n\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)\n\tat org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)\n\tat org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:684)\n\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)\n\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:502)\n\tat org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1152)\n\tat org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)\n\tat org.apache.tomcat.util.net.AprEndpoint$SocketWithOptionsProcessor.run(AprEndpoint.java:2464)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)\n\tat java.lang.Thread.run(Thread.java:748)\n""}`

**Comment by umbraesoulsbane** [comment]
That's the same error.

Are you sure both containers are running? Can you connect to the DB through MySql Workbench? If so were any tables added?

You can shell into the Mura container and test whether you can ping the DB which should be svc_muradb. You can also try to telnet to svc_muradb port 3306, but you would have to install telnet which isn't in the base image. If you can reach the DB from Mura then something has to be amiss with the DB. Assuming you are running without changing the yml.

**Comment by SowjiChenji** [comment]
I cannot connect to the DB from MySql Workbench nor telnet into ports 3306. I think there is something wrong with the DB. 

`  svc_muradb:
    image: mysql:latest
    environment:
      MYSQL_DATABASE: muradb
      MYSQL_ROOT_PASSWORD: 5trongP@55w0rd
    volumes:
      - vol_muradb:/var/lib/mysql
    ports:
      - "5001:3306"`

**Comment by SowjiChenji** [comment]
Just to be sure, may I know what is the server name to use on MySql Workbench with the default .yml file? @umbraesoulsbane

**Comment by umbraesoulsbane** [comment]
You would need to connect to localhost on port 5001. The ports section of the yml is formatted HOST:CONTAINER. 

Did a ping work? Did telnet timeout or connection refused? Do you have errors in the docker logs for the MySQL container? Can you shell into the DB container?
 
Sounds like this is an issue with MySQL, maybe that specific version, running on your system. Maybe try one of the other examples? I tend to prefer PostgreSQL.

**Comment by umbraesoulsbane** [comment]
@SowjiChenji 

I decided to publish my personal layout for dockerizing MuraCMS. One used MySQL and I was able to reproduce your problem with a fresh launch and image pull. In my case, I could connect outside of the container and the MuraCMS container could reach the svc_muradb service and the port. However, Mura/Lucee reported the connection failure and the DB tables were not created. I could see open connections from the MuraCMS container, but these were all sleeping and seemed to be what I would consider post install selects.

I wish I saved the search but I found several references to an issue with MySQL where using a persistent volume and JDBC connect string can lead to grants not being setup properly at start up. This may be related to a specific MySQL or Docker version. I did not confirm this, but you might need to try and connect to MySQL and double check the user and permissions necessary for Mura to connect. This may need to be manually setup. You might also be able to remove the JDBC connection string and do a more native connection (specifically lines 17 & 18 in the yml). I did not have these lines in the project fork I was using which was the version I got working.


If you would like to try my project it is at:
https://github.com/umbraesoulsbane/muracms-docked

This uses PostgresSQL instead and uses Dockerfiles to allow more control over the MuraCMS container for adding debug tools and such. It is more complicated than the BlueRiver examples, but should bring up a blank instance smoothly by default. This is intended for developing with Mura and not specifically for running a production instance. Although it can be hardened and uses Docker secrets to be converted for use in production.

Good luck.
